maven-artifact/src/main/java/org/apache/maven/artifact/versioning/VersionRange.java

```diff
@@ -227,9 +223,13 @@ private static Restriction parseRestriction( String spec )
                 upperVersion = new DefaultArtifactVersion( upperBound );
             }
 
-            if ( upperVersion != null && lowerVersion != null && upperVersion.compareTo( lowerVersion ) < 0 )
+            if ( upperVersion != null && lowerVersion != null )
             {
-                throw new InvalidVersionSpecificationException( "Range defies version ordering: " + spec );
+                int result = upperVersion.compareTo( lowerVersion );
+                if ( result < 0 || ( result == 0 && ( !lowerBoundInclusive || !upperBoundInclusive ) ) )
+                {
+                    throw new InvalidVersionSpecificationException( "Range defies version ordering: " + spec );
+                }
```

Comment:
Sorry, I meant this change. Since the condition has changed maybe a test should cover this as well, if none does?

Reply:
There was and still is a test covering this: https://github.com/apache/maven/blob/c8518e811f53c4fe89a85616aa3ff94e37a91e77/maven-artifact/src/test/java/org/apache/maven/artifact/versioning/VersionRangeTest.java#L177-L178

Reply:
Thanks, this is all a wanted to see. Let me test this...